internal/provider/clickhouse_cluster_resource.go

```diff
@@ -56,11 +58,10 @@ func (m *clickhouseClusterResources) convert() (*clickhouse.ClusterResources, di
 
 	if v := m.Clickhouse; v != nil {
 		r.Clickhouse = &clickhouse.ClusterResources_Clickhouse{}
-		if v := m.Clickhouse.ResourcePresetId; !v.IsNull() {
-			r.Clickhouse.ResourcePresetId = v.ValueString()
-		} else {
-			diags.AddError("missed resource_preset_id", "clickhouse resource_preset_id must be set")
-		}
+		r.Clickhouse.ResourcePresetId = m.Clickhouse.ResourcePresetId.ValueString()
```

Comment:
I don't remember exactly but as I understand it's important to use these conditions.
Otherwise we could nullify some fields in cluster if user didn't describe them in the model (terraform state).
So, for example, user could create cluster with autoscaling from UI and forgot to mentioned it in TF and next `tf apply` will drop autoscaling settings.

Reply:
It is validated in clusterResourcesValidator, so here we sure that it's filled or empty if autoscaler is enabled